library/tlconfig.py

```diff
@@ -55,17 +58,24 @@ def main():
     if len(module.params['value']) == 0:
         module.fail_json(msg="Value can't be empty")
 
-    if getvalue(module.params['param']) != module.params['value']:
+    current=getvalue(module.params['param'])
+    new=module.params['value']
+
+    if current != new:
         ret = setvalue(module.params['param'], module.params['value'])
 
         if ret is not 0:
             msg = "Failed setting %s: tl-config returned %d" % (module.params['param'], ret)
             module.fail_json(msg=msg)
-        changed = True
+        result['changed'] = True
+
+        if module._diff:
+            result['diff'] = dict(
+                before=yaml.safe_dump(current),
+                after=yaml.safe_dump(new)
+            )
 
-    module.exit_json(changed=changed,
-                     param=module.params['param'],
-                     value=module.params['value'])
+    module.exit_json(**result)
```

Comment:
Yes, I didn't include results in the original post, but here's the output from diff:

```
TASK [thinlinc_server : Configure /vsmagent/agent_hostname] *********************************************************
--- before
+++ after
@@ -1,2 +1,2 @@
-thinlinc.zm1.element-networks.nl
+thinlinc.element-networks.nl
 ...

changed: [thinlinc.zm1.element-networks.nl] => changed=true 
```

The main issue I see is that it can be a bit confusing (the diff output isn't the prettiest)

```
TASK [thinlinc_server : Configure /vsmagent/agent_hostname] *********************************************************
--- before
+++ after
@@ -1,2 +1,2 @@
-thinlinc.zm1.element-networks.nl
+thinlinc.element-networks.nl
 ...

changed: [thinlinc.zm1.element-networks.nl] => changed=true 
  param: /vsmagent/agent_hostname
  value: thinlinc.element-networks.nl
```

I put them back in for now